### PR TITLE
Fix handling the Vault PKI certificate chain

### PR DESCRIPTION
**Pull request by hiyosi**
Signed-off-by: Tomoya Usami <tousami@zlab.co.jp>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

UpstreamAuthority `spire` plugin

**Description of change**

This fixes that requiring a Vault PKI certificate when connecting to SPIRE Server if the VaultPKI secret engine is configured as an intermediate CA.
Before the fix, the X509CA certificate was not chained and all Upstream CAs, including the root, were treated as bundles.

**Which issue this PR fixes**

### CODE REVIEW

**Comment by jkevlin** [comment]
Which Issue does this address?
How many intermediate certs does this include in the chain?

**Comment by hiyosi** [comment]
@jkevlin 
This is not tied to any issue. 
I confirmed in my environment.

```
    External CA   --- (a)
       |
    Vault PKI     --- (b)
       |
    SPIRE CA      --- (c)
       |
    SPIRE Server  --- (d)
       ^
       |
     Agent
```     


Before this PR, the SPIRE Server serviced (c) , (d).
But when the Vault PKI configured as Intermediate as (b), the SPIRE Agent needs (a), (b) as bootstap trust bundles.
After fixes, the SPIRE Server's certificate chain inclueds any intermediate certificate which necessary to chain back to the Root CA. I think this is the same behavior as other plugins (e.g. awspca).

**Comment by jkevlin** [comment]
It looks like you have chosen to include Root_CA in the bundle.  I know that is an option it also adds to the size of the payload.
What is your thinking on that?
      certificate_list
      This is a sequence (chain) of certificates.  The sender's
      certificate MUST come first in the list.  Each following
      certificate MUST directly certify the one preceding it.  Because
      certificate validation requires that root keys be distributed
      independently, the self-signed certificate that specifies the root
      certificate authority MAY be omitted from the chain, under the
      assumption that the remote end must already possess it in order to
      validate it in any case.

**Comment by evan2645** [comment]
Thank you for sending this @hiyosi - I have not reviewed the code yet, but I think what you're describing sounds right

> But when the Vault PKI configured as Intermediate

Out of curiosity, is this the default Vault behavior (to create an intermediate off of which it issues requested certs)? Or is it that the Vault PKI has been loaded with an intermediate, and the root is stored/managed elsewhere? Or something else?

> It looks like you have chosen to include Root_CA in the bundle. I know that is an option it also adds to the size of the payload.
> What is your thinking on that?

👋 Hi @jkevlin :)

I'm not sure if I exactly follow you here, but I _think_ your question is around the tradeoff of including (b) in the bundle vs appending it to the chain. SPIRE has a long history of attempts to cut chains short, store intermediate certificates in bundles, etc... All those attempts ended in pain and suffering, and we have spent some time cleaning it all up.

The long story short is that cutting the chain short and/or distributing intermediates in the bundle can lead to surprising behavior. There is a general assumption amongst X.509 validators and TLS implementations that the chain will be complete, and that certificates in the bundle are self-signed.

> certificate_list

Reference: https://tools.ietf.org/html/rfc5246#section-7.4.2

My understanding is that the proposed fix brings us in line with the language in RFC 5246 section 7.4.2.

**Comment by jkevlin** [comment]
@evan2645 it wasn't clear to me, if this add a) the root_ca to the chain.  If so, I was just wondering what the logic was?

**Comment by hiyosi** [comment]
@evan2645 

> Out of curiosity, is this the default Vault behavior (to create an intermediate off of which it issues requested certs)? Or is it that the Vault PKI has been loaded with an intermediate, and the root is stored/managed elsewhere? Or something else?

Vault can be configured as both an Root CA or an Intermediate CA. It is also possible to provide Root and Intermediate CAs at  different mount points within a single Vault instance. 

c.f. https://www.vaultproject.io/docs/secrets/pki#quick-start

**Comment by jkevlin** [comment]
Our implementation for vault is that it is a 3 tier design root->intermediate->issuing_ca and the chain that we include is for the intermediate and the issuing_ca, knowing that our root _ca is distributed ahead of time.  logically if you include the root_ca and use that as the basis for your chain of trust, you don't really have any trust, any one could slip a complete chain in there. I always assumed the root_ca would be delivered out of band and inclusion in the chain was redundant. It is not a big deal, just curious why it was included.

**Review by amartinezfayo — APPROVED** [review]
Thank you @hiyosi for this. The code changes look good to me.
As @evan2645 pointed, the important consideration for this fix is that is aligned with RFC 5246 section 7.4.2. Diverging from that has been proven to be problematic.

**Comment by evan2645** [comment]
> @evan2645 it wasn't clear to me, if this add a) the root_ca to the chain. If so, I was just wondering what the logic was?

I _think_ we're saying the same thing? Previously, this plugin included A and B in the bundle, with only C in the chain. This change makes it so that only A is in the bundle, and B and C are in the chain.

I don't have a vault deployment handy to test, but a quick review of the code and that's what it looks like to me.

**Comment by hiyosi** [comment]
> Previously, this plugin included A and B in the bundle, with only C in the chain. This change makes it so that only A is in the bundle, and B and C are in the chain.

Thank you! that's correct.

## Before
```
opt/spire # bin/spire-agent api fetch x509 -write . -socketPath /run/spire/sockets/agent.sock
Received 1 svid after 16.4566ms

SPIFFE ID:              spiffe://example.org/my-app
SVID Valid After:       2020-12-11 00:09:15 +0000 UTC
SVID Valid Until:       2020-12-11 00:10:25 +0000 UTC
Intermediate #1 Valid After:    2020-12-11 00:04:01 +0000 UTC
Intermediate #1 Valid Until:    2020-12-11 00:14:31 +0000 UTC
CA #1 Valid After:      2020-12-11 00:00:41 +0000 UTC
CA #1 Valid Until:      2021-12-11 00:01:11 +0000 UTC
CA #2 Valid After:      2020-12-10 23:58:46 +0000 UTC
CA #2 Valid Until:      2021-12-10 23:59:16 +0000 UTC

Writing SVID #0 to file svid.0.pem.
Writing key #0 to file svid.0.key.
Writing bundle #0 to file bundle.0.pem.

/opt/spire # ls
bin           bundle.0.pem  svid.0.key    svid.0.pem
```
#### Trust Bundles
```
/opt/spire # cat bundle.0.pem
-----BEGIN CERTIFICATE-----
MIIDOzCCAiOgAwIBAgIUIBHv4rtOb6U8TzP0gS14nY7z2BwwDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYS5leGFtcGxlLm9yZzAeFw0yMDEyMTEwMDAwNDFaFw0y
MTEyMTEwMDAxMTFaMBgxFjAUBgNVBAMTDWIuZXhhbXBsZS5vcmcwggEiMA0GCSqG
SIb3DQEBAQUAA4IBDwAwggEKAoIBAQC+bUopr/H/sETkm+jnJ+kMip0Ir6vSDM7F
hsz0T4OaUANI5FNwoZqn3Nz/u6Ygk5/sd1M71euVJ/cYf3hs1RLawaFEx+29XLYn
wG4FfTN84NyH/JA/wv//Uwj+4Ei/e86pjPlodPa8sgDoUeYZ+gteZRecy/NRH1BF
kuIOLR6zUDiLG+0tO8rnJKMOaLILKAINm66okMXg1miBNCO/AgVsqCXAbhm7AyGZ
H+Fgmb+LEVAYsdW1Tr6Rs+5bENLrXBW5CRfN72qUeuAODfBalwEh/EFtQDbcHsMs
0bu5hUDMP/D3IUZvYY2IPRCIJsQVpORkspG2/dyz3wFWTs9dA50/AgMBAAGjfTB7
MA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBQsYtlQ
JlQ03hFhdLdkTC3jnx0UrDAfBgNVHSMEGDAWgBTDoDVm2Xhl3huIVyJXqDZbRA6Z
2zAYBgNVHREEETAPgg1iLmV4YW1wbGUub3JnMA0GCSqGSIb3DQEBCwUAA4IBAQBp
9Tc5JulqHRC99zzncn3ZdYSmuG08UylvCZDv8nP4YPzqaYBqyOMg1+SjlFBNk3jw
3s3DYCehPoMY5uc3GUWdLjLRYeRfBbG2jGWkzxQJvbGy6Dd7Dt4t/LyDPqm3tuHf
MbNkyQ9hKJZ6vVCZX64QlrCTzNM9IrfavjuYUSh2F0b0IOlK1YVfXv19VvVRZOPH
hw/ABBKmH6h4ZTrSCS2d+5SQQNtm2ZXIut+QbyDd4bZfVX3k7bNYp07qP029S5EC
zxHTU/moqfFqJuHn3xdnkZS4GXBnv7SjJSMCFGOodpQv9gcD6/pW7LiBZyXoUdO5
b40RzsLZtISNGQN9pdhn
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIDOzCCAiOgAwIBAgIUCJJbv1AwocsD8K+9+oXXgroSysgwDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYS5leGFtcGxlLm9yZzAeFw0yMDEyMTAyMzU4NDZaFw0y
MTEyMTAyMzU5MTZaMBgxFjAUBgNVBAMTDWEuZXhhbXBsZS5vcmcwggEiMA0GCSqG
SIb3DQEBAQUAA4IBDwAwggEKAoIBAQDRkilexb8i3tnaaSzSRaGEgPP8k7qTI+Jm
MVaOJNNL5/XD4TvuZmAXyjfkL21FszCjoDVDpffZ8TQIZoSCjBgAU19V53F8SP2t
sM0ufHpx8sF7hRu3+vCRw+mHtKLYG2Ovz2FTjlbwNSXYFPnK2u/46AxbzNMHikgB
J83p0oLLac5gqC6oSKwdtiNf0+ZGnBZ+TZskQEx7qBOJ2egmkAsR11oQbOX+SdxA
tXLpIQN4BNCRfsoTpZi55S5cur7wrwxmGK7QnK/0qforL1q/yGEUn/Qo/L8vbE6F
RFrDTxQaa/CZl4aChDNAAu+2aVcTcY93u2piEEICKYntQHBsmqrDAgMBAAGjfTB7
MA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBTDoDVm
2Xhl3huIVyJXqDZbRA6Z2zAfBgNVHSMEGDAWgBTDoDVm2Xhl3huIVyJXqDZbRA6Z
2zAYBgNVHREEETAPgg1hLmV4YW1wbGUub3JnMA0GCSqGSIb3DQEBCwUAA4IBAQA4
lWIY8l4d74qMLo9xUY1sy1Wu0/lYn8yZ/rO1R5YKqsRg1JF6GE3GGdyrJThv/gcu
IMvz5ysoYmHEO9gmuXQhpFqaJxM9mXlQquKvTpMMT/tMvt7dqCjgoCtd9Yf8S0Ii
WqDegfDN6agPVNvPbvlr+SqOQqL18wqei14BN5u+fEOc1VLlVjniHM4W/4dOQ3T/
nSLus4zVl7kSAYq70olJYPRq6DxWoOeYMAboqd6XAmKA0wTiFOw5NAWEXqEqYVXD
j01tOJ8Lm1BISdeXYOAxIaCD4g9sEHFEbKBEyj4qdbgEJ3THQBLQSTlK1ZhXkQJA
sHdhJNZvFl4c/XqBS5bE
-----END CERTIFICATE-----
```
#### SVID
```
/opt/spire # cat svid.0.pem
-----BEGIN CERTIFICATE-----
MIIB4zCCAYmgAwIBAgIRALjBEiEeUGyXlckK2cWlE/swCgYIKoZIzj0EAwIwHjEL
MAkGA1UEBhMCVVMxDzANBgNVBAoTBlNQSUZGRTAeFw0yMDEyMTEwMDA5MTVaFw0y
MDEyMTEwMDEwMjVaMB0xCzAJBgNVBAYTAlVTMQ4wDAYDVQQKEwVTUElSRTBZMBMG
ByqGSM49AgEGCCqGSM49AwEHA0IABJr6kVokuY00hLyVUk/GDEAbxn5sDijxPYYW
OYiLJrfX4NsK/xhrbNFCn0tmd7eltkOXUPUvMdxM4YRs2GcTOEujgagwgaUwDgYD
VR0PAQH/BAQDAgOoMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAMBgNV
HRMBAf8EAjAAMB0GA1UdDgQWBBR4EyjPME7UHEtkKRmS1/NuNyQGDDAfBgNVHSME
GDAWgBRKmLtRIDej7MG+5n2sizV8ylM/TTAmBgNVHREEHzAdhhtzcGlmZmU6Ly9l
eGFtcGxlLm9yZy9teS1hcHAwCgYIKoZIzj0EAwIDSAAwRQIgAVMtcax4cLNdecEg
arwlVxjleXhArr/3d4ApzpQIkzcCIQChicnqcn2utkBIyRM5yPycKeeq2qyGRfjE
5Fl+IVhnMQ==
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIICXDCCAUSgAwIBAgIUONPid8mBn7mec1Y6DDr4x6hXE6cwDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYi5leGFtcGxlLm9yZzAeFw0yMDEyMTEwMDA0MDFaFw0y
MDEyMTEwMDE0MzFaMB4xCzAJBgNVBAYTAlVTMQ8wDQYDVQQKEwZTUElGRkUwWTAT
BgcqhkjOPQIBBggqhkjOPQMBBwNCAARKMg8yysgVbHmwzh26wnEOSR6SiBge1Q1b
iSapSkSEMVz/x0oUDtj2ITHoQr6XOO1iYVVl1R661+a9bnd1L7Ulo2MwYTAOBgNV
HQ8BAf8EBAMCAQYwDwYDVR0TAQH/BAUwAwEB/zAdBgNVHQ4EFgQUSpi7USA3o+zB
vuZ9rIs1fMpTP00wHwYDVR0jBBgwFoAULGLZUCZUNN4RYXS3ZEwt458dFKwwDQYJ
KoZIhvcNAQELBQADggEBAEVnKJ+2M/4BvxroFqixO1hFW5RSdjoQUlXffhpSYj6e
/oTQVymU4FqxGE5uwZlQFthF+1ojqWU7zpyWdJqd+DdzOvd9JaDHBPJay32/V5P4
fuPrvD4Y89LtvWtkszBDi7PbgNeOCewFNNIHUYhHbIJfR+ulKRGYZm0Dt3X6mx2P
55GKSjdsKNofZJgl9Qxy+Sr6OJ3kWke39t6F8ZqosBhdW81ikRVubKykKyRPCUls
nzjVC5GfoVhjGius8jhAol52VuqOv84l58vfwBL1w0SPmqoDM02e0Rxa8o0R0u+S
QFiM2GCsAbicLVTbPuL0/9ZSGlZfAjch1/Rfyj252X8=
-----END CERTIFICATE-----
```

## After

```
/opt/spire # bin/spire-agent api fetch x509 -write . -socketPath /run/spire/sockets/agent.sock
Received 1 svid after 13.5716ms

SPIFFE ID:              spiffe://example.org/my-app
SVID Valid After:       2020-12-11 00:15:56 +0000 UTC
SVID Valid Until:       2020-12-11 00:17:06 +0000 UTC
Intermediate #1 Valid After:    2020-12-11 00:13:45 +0000 UTC
Intermediate #1 Valid Until:    2020-12-11 00:24:15 +0000 UTC
Intermediate #2 Valid After:    2020-12-11 00:00:41 +0000 UTC
Intermediate #2 Valid Until:    2021-12-11 00:01:11 +0000 UTC
CA #1 Valid After:      2020-12-10 23:58:46 +0000 UTC
CA #1 Valid Until:      2021-12-10 23:59:16 +0000 UTC

Writing SVID #0 to file svid.0.pem.
Writing key #0 to file svid.0.key.
Writing bundle #0 to file bundle.0.pem.
/opt/spire #
/opt/spire # ls
bin           bundle.0.pem  svid.0.key    svid.0.pem
```
#### Trust Bundles
```
/opt/spire # cat bundle.0.pem
-----BEGIN CERTIFICATE-----
MIIDOzCCAiOgAwIBAgIUCJJbv1AwocsD8K+9+oXXgroSysgwDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYS5leGFtcGxlLm9yZzAeFw0yMDEyMTAyMzU4NDZaFw0y
MTEyMTAyMzU5MTZaMBgxFjAUBgNVBAMTDWEuZXhhbXBsZS5vcmcwggEiMA0GCSqG
SIb3DQEBAQUAA4IBDwAwggEKAoIBAQDRkilexb8i3tnaaSzSRaGEgPP8k7qTI+Jm
MVaOJNNL5/XD4TvuZmAXyjfkL21FszCjoDVDpffZ8TQIZoSCjBgAU19V53F8SP2t
sM0ufHpx8sF7hRu3+vCRw+mHtKLYG2Ovz2FTjlbwNSXYFPnK2u/46AxbzNMHikgB
J83p0oLLac5gqC6oSKwdtiNf0+ZGnBZ+TZskQEx7qBOJ2egmkAsR11oQbOX+SdxA
tXLpIQN4BNCRfsoTpZi55S5cur7wrwxmGK7QnK/0qforL1q/yGEUn/Qo/L8vbE6F
RFrDTxQaa/CZl4aChDNAAu+2aVcTcY93u2piEEICKYntQHBsmqrDAgMBAAGjfTB7
MA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBTDoDVm
2Xhl3huIVyJXqDZbRA6Z2zAfBgNVHSMEGDAWgBTDoDVm2Xhl3huIVyJXqDZbRA6Z
2zAYBgNVHREEETAPgg1hLmV4YW1wbGUub3JnMA0GCSqGSIb3DQEBCwUAA4IBAQA4
lWIY8l4d74qMLo9xUY1sy1Wu0/lYn8yZ/rO1R5YKqsRg1JF6GE3GGdyrJThv/gcu
IMvz5ysoYmHEO9gmuXQhpFqaJxM9mXlQquKvTpMMT/tMvt7dqCjgoCtd9Yf8S0Ii
WqDegfDN6agPVNvPbvlr+SqOQqL18wqei14BN5u+fEOc1VLlVjniHM4W/4dOQ3T/
nSLus4zVl7kSAYq70olJYPRq6DxWoOeYMAboqd6XAmKA0wTiFOw5NAWEXqEqYVXD
j01tOJ8Lm1BISdeXYOAxIaCD4g9sEHFEbKBEyj4qdbgEJ3THQBLQSTlK1ZhXkQJA
sHdhJNZvFl4c/XqBS5bE
-----END CERTIFICATE-----
```
#### SVID
```
/opt/spire # cat svid.0.pem
-----BEGIN CERTIFICATE-----
MIIB5DCCAYmgAwIBAgIRAOQHXV7DYY8snURPksMuiGIwCgYIKoZIzj0EAwIwHjEL
MAkGA1UEBhMCVVMxDzANBgNVBAoTBlNQSUZGRTAeFw0yMDEyMTEwMDE1NTZaFw0y
MDEyMTEwMDE3MDZaMB0xCzAJBgNVBAYTAlVTMQ4wDAYDVQQKEwVTUElSRTBZMBMG
ByqGSM49AgEGCCqGSM49AwEHA0IABPq+8Kn9GI3c7/DRI9sc8YL95Cw+Hi7byinB
px/odgiVPSQ+CpB/dc3AJ/0oTk+FwMJ63SrWIrTHjM0aj0rwmvWjgagwgaUwDgYD
VR0PAQH/BAQDAgOoMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAMBgNV
HRMBAf8EAjAAMB0GA1UdDgQWBBT2EX1U2q7379pun2IYkp7P8KhI2zAfBgNVHSME
GDAWgBTQ7POLZIkt0ZbdmKCBIOIAQPlDpzAmBgNVHREEHzAdhhtzcGlmZmU6Ly9l
eGFtcGxlLm9yZy9teS1hcHAwCgYIKoZIzj0EAwIDSQAwRgIhAPT6TmeYq9guzbKf
OpQ09qL+dQYBilr/4dFp3G0cwrErAiEAlvX0HxoMEtuhbCayawa+yNv22f0B35HL
gE0esd0z89k=
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIICXDCCAUSgAwIBAgIUXvWgZBJiKZUzI3tGLBA5FSnuJqowDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYi5leGFtcGxlLm9yZzAeFw0yMDEyMTEwMDEzNDVaFw0y
MDEyMTEwMDI0MTVaMB4xCzAJBgNVBAYTAlVTMQ8wDQYDVQQKEwZTUElGRkUwWTAT
BgcqhkjOPQIBBggqhkjOPQMBBwNCAAQX49jl92L14S+pfU4WtdnSqqRFb9ASD5Ll
d2bYn76cJC6zur9125GKzX+BOBPSLlu2P0aQFfsceOE8XsrRMyp0o2MwYTAOBgNV
HQ8BAf8EBAMCAQYwDwYDVR0TAQH/BAUwAwEB/zAdBgNVHQ4EFgQU0Ozzi2SJLdGW
3ZiggSDiAED5Q6cwHwYDVR0jBBgwFoAULGLZUCZUNN4RYXS3ZEwt458dFKwwDQYJ
KoZIhvcNAQELBQADggEBAH3Z5utXxUfl+eVCWLV1tLiWHoks5dE6fApR/pnKLqCF
eTrv+PkM2NtHnd1mhXjYLuulBLsktv/S8Hwf6YOj4Sq3AIdsenRUoSOBONxXURjc
dXUIFhQI1VOvHbMNWSjTgt2sRfU89XJlicErp6MFkO6177O1BB6+6JXj8W+8fzFe
E7UMa360T1G+wLzvTsJH8YrZ1ZipzbC+pzelFkB9OjIjHhICU255qbtjrsL1nA5y
VM3HhvFRwuRzPmhNj4yMpB8EFFA4daO52vWMwHTJwj6OHUU+GE6BXGSZFGSFjOma
OJUkQq4zrdrbs6b30vSqKHz837On0ygnPjZrkI1vc9U=
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIDOzCCAiOgAwIBAgIUIBHv4rtOb6U8TzP0gS14nY7z2BwwDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYS5leGFtcGxlLm9yZzAeFw0yMDEyMTEwMDAwNDFaFw0y
MTEyMTEwMDAxMTFaMBgxFjAUBgNVBAMTDWIuZXhhbXBsZS5vcmcwggEiMA0GCSqG
SIb3DQEBAQUAA4IBDwAwggEKAoIBAQC+bUopr/H/sETkm+jnJ+kMip0Ir6vSDM7F
hsz0T4OaUANI5FNwoZqn3Nz/u6Ygk5/sd1M71euVJ/cYf3hs1RLawaFEx+29XLYn
wG4FfTN84NyH/JA/wv//Uwj+4Ei/e86pjPlodPa8sgDoUeYZ+gteZRecy/NRH1BF
kuIOLR6zUDiLG+0tO8rnJKMOaLILKAINm66okMXg1miBNCO/AgVsqCXAbhm7AyGZ
H+Fgmb+LEVAYsdW1Tr6Rs+5bENLrXBW5CRfN72qUeuAODfBalwEh/EFtQDbcHsMs
0bu5hUDMP/D3IUZvYY2IPRCIJsQVpORkspG2/dyz3wFWTs9dA50/AgMBAAGjfTB7
MA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBQsYtlQ
JlQ03hFhdLdkTC3jnx0UrDAfBgNVHSMEGDAWgBTDoDVm2Xhl3huIVyJXqDZbRA6Z
2zAYBgNVHREEETAPgg1iLmV4YW1wbGUub3JnMA0GCSqGSIb3DQEBCwUAA4IBAQBp
9Tc5JulqHRC99zzncn3ZdYSmuG08UylvCZDv8nP4YPzqaYBqyOMg1+SjlFBNk3jw
3s3DYCehPoMY5uc3GUWdLjLRYeRfBbG2jGWkzxQJvbGy6Dd7Dt4t/LyDPqm3tuHf
MbNkyQ9hKJZ6vVCZX64QlrCTzNM9IrfavjuYUSh2F0b0IOlK1YVfXv19VvVRZOPH
hw/ABBKmH6h4ZTrSCS2d+5SQQNtm2ZXIut+QbyDd4bZfVX3k7bNYp07qP029S5EC
zxHTU/moqfFqJuHn3xdnkZS4GXBnv7SjJSMCFGOodpQv9gcD6/pW7LiBZyXoUdO5
b40RzsLZtISNGQN9pdhn
-----END CERTIFICATE-----
```

## Vault PKI
```
vault read pki_int/cert/ca_chain
Key                Value
---                -----
certificate        -----BEGIN CERTIFICATE-----
MIIDOzCCAiOgAwIBAgIUIBHv4rtOb6U8TzP0gS14nY7z2BwwDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYS5leGFtcGxlLm9yZzAeFw0yMDEyMTEwMDAwNDFaFw0y
MTEyMTEwMDAxMTFaMBgxFjAUBgNVBAMTDWIuZXhhbXBsZS5vcmcwggEiMA0GCSqG
SIb3DQEBAQUAA4IBDwAwggEKAoIBAQC+bUopr/H/sETkm+jnJ+kMip0Ir6vSDM7F
hsz0T4OaUANI5FNwoZqn3Nz/u6Ygk5/sd1M71euVJ/cYf3hs1RLawaFEx+29XLYn
wG4FfTN84NyH/JA/wv//Uwj+4Ei/e86pjPlodPa8sgDoUeYZ+gteZRecy/NRH1BF
kuIOLR6zUDiLG+0tO8rnJKMOaLILKAINm66okMXg1miBNCO/AgVsqCXAbhm7AyGZ
H+Fgmb+LEVAYsdW1Tr6Rs+5bENLrXBW5CRfN72qUeuAODfBalwEh/EFtQDbcHsMs
0bu5hUDMP/D3IUZvYY2IPRCIJsQVpORkspG2/dyz3wFWTs9dA50/AgMBAAGjfTB7
MA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBQsYtlQ
JlQ03hFhdLdkTC3jnx0UrDAfBgNVHSMEGDAWgBTDoDVm2Xhl3huIVyJXqDZbRA6Z
2zAYBgNVHREEETAPgg1iLmV4YW1wbGUub3JnMA0GCSqGSIb3DQEBCwUAA4IBAQBp
9Tc5JulqHRC99zzncn3ZdYSmuG08UylvCZDv8nP4YPzqaYBqyOMg1+SjlFBNk3jw
3s3DYCehPoMY5uc3GUWdLjLRYeRfBbG2jGWkzxQJvbGy6Dd7Dt4t/LyDPqm3tuHf
MbNkyQ9hKJZ6vVCZX64QlrCTzNM9IrfavjuYUSh2F0b0IOlK1YVfXv19VvVRZOPH
hw/ABBKmH6h4ZTrSCS2d+5SQQNtm2ZXIut+QbyDd4bZfVX3k7bNYp07qP029S5EC
zxHTU/moqfFqJuHn3xdnkZS4GXBnv7SjJSMCFGOodpQv9gcD6/pW7LiBZyXoUdO5
b40RzsLZtISNGQN9pdhn
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIDOzCCAiOgAwIBAgIUCJJbv1AwocsD8K+9+oXXgroSysgwDQYJKoZIhvcNAQEL
BQAwGDEWMBQGA1UEAxMNYS5leGFtcGxlLm9yZzAeFw0yMDEyMTAyMzU4NDZaFw0y
MTEyMTAyMzU5MTZaMBgxFjAUBgNVBAMTDWEuZXhhbXBsZS5vcmcwggEiMA0GCSqG
SIb3DQEBAQUAA4IBDwAwggEKAoIBAQDRkilexb8i3tnaaSzSRaGEgPP8k7qTI+Jm
MVaOJNNL5/XD4TvuZmAXyjfkL21FszCjoDVDpffZ8TQIZoSCjBgAU19V53F8SP2t
sM0ufHpx8sF7hRu3+vCRw+mHtKLYG2Ovz2FTjlbwNSXYFPnK2u/46AxbzNMHikgB
J83p0oLLac5gqC6oSKwdtiNf0+ZGnBZ+TZskQEx7qBOJ2egmkAsR11oQbOX+SdxA
tXLpIQN4BNCRfsoTpZi55S5cur7wrwxmGK7QnK/0qforL1q/yGEUn/Qo/L8vbE6F
RFrDTxQaa/CZl4aChDNAAu+2aVcTcY93u2piEEICKYntQHBsmqrDAgMBAAGjfTB7
MA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBTDoDVm
2Xhl3huIVyJXqDZbRA6Z2zAfBgNVHSMEGDAWgBTDoDVm2Xhl3huIVyJXqDZbRA6Z
2zAYBgNVHREEETAPgg1hLmV4YW1wbGUub3JnMA0GCSqGSIb3DQEBCwUAA4IBAQA4
lWIY8l4d74qMLo9xUY1sy1Wu0/lYn8yZ/rO1R5YKqsRg1JF6GE3GGdyrJThv/gcu
IMvz5ysoYmHEO9gmuXQhpFqaJxM9mXlQquKvTpMMT/tMvt7dqCjgoCtd9Yf8S0Ii
WqDegfDN6agPVNvPbvlr+SqOQqL18wqei14BN5u+fEOc1VLlVjniHM4W/4dOQ3T/
nSLus4zVl7kSAYq70olJYPRq6DxWoOeYMAboqd6XAmKA0wTiFOw5NAWEXqEqYVXD
j01tOJ8Lm1BISdeXYOAxIaCD4g9sEHFEbKBEyj4qdbgEJ3THQBLQSTlK1ZhXkQJA
sHdhJNZvFl4c/XqBS5bE
-----END CERTIFICATE-----
revocation_time    0
```

**Comment by amartinezfayo** [comment]
Thank you @hiyosi for providing an example of the behavior _before_ and _after_ this change. That confirms the interpretation of the code changes.
I'm merging this.
